### ISSUE TITLE
Wrap QApplication and other major application objects in their own class

### ISSUE THREAD

**Issue by heartsucker** [freedomofpress/securedrop-client#314]
We have some objects like the `QApplication`, GUI (`Window`), controller (`Client`) and some others that are "master" objects for a given application process. These are "free floating" within the `securedrop_client.app.init_app()` function and never get bound together in a sensible way. In order to be able to talk about "one running instance of the application" we should wrap these into a class called `ClientApp` to keep all the major pieces in the same place and to give us the vocabulary to talk about the whole application at once.

In short, `securedrop_client.app.init` becomes `securedrop_client.app.ClientApp.__init__` and the `ClientApp` is really just an algebraic data type.

**Comment by creviera** [comment]
I think it would help before making a high-level refactor to create an architecture diagram of the client with all its current components. Right now, I think we have the following high-level objects:

**Comment by creviera** [comment]
https://github.com/freedomofpress/securedrop-client/wiki/Architecture was started since this opened but this issue needs more discussion.

**Comment by eloquence** [comment]
Closing all refactoring issues pertaining to the old codebase since we're focused on the Electron rewrite.
